arroyo/processing/strategies/dead_letter_queue/policies/abstract.py

```diff
@@ -8,11 +9,17 @@ class InvalidMessages(Exception):
     so they are handled correctly.
     """
 
-    def __init__(self, messages: Sequence[Any]):
+    def __init__(self, messages: Sequence[Any], topic: Optional[str] = None):
```

Comment:
Why is topic optional? Under what circumstances would it be present or not present?

Reply:
This topic is meant to be the topic the message(s) were originally produced to. From our discussion, I think we could throw almost anything to the DLQ, so I'm guessing it's possible at some point we no longer know what the original message or topic was? Eg if there were some bad clickhouse rows, I'm not certain we would have the original topic a message was produced to to trigger this down the line.

I can make it required if you think it will always be known, I'm just trying to keep it open to any use case.